### ISSUE TITLE
Opus files not generated properly

### ISSUE THREAD

**Issue by k3x** [SFTtech/openage#1068]
```~/Desktop/openage$ ./run
INFO [py] launching openage v0.3.0-1331-g0def3d32
INFO [py] compiled by GNU 7.3.0
INFO [py] running in DEVMODE
INFO launching engine with Path(Union(Directory(/home/felix/Desktop/openage/cfg).root @ (b'cfg',), Directory(/home/felix/Desktop/openage/assets).root @ (b'assets',)):) and fps limit 0
INFO SDL audio subsystems initialized
INFO Using audio device: default [freq=48000, format=32784, channels=2, samples=4096]
INFO Initialized SDL video subsystems.
INFO Compiled with Qt 5.9.5 and run with Qt 5.9.5
INFO loading configuration files...
INFO [py] loading config file [Union(Directory(/home/felix/Desktop/openage/cfg).root @ (b'cfg',), Directory(/home/felix/Desktop/openage/assets).root @ (b'assets',))]:cfg/keybinds.oac...
INFO Loading time [engine]: 0.387 s
INFO Loading time   [game]: 0.403 s
INFO [T1] Loading game specification files...
INFO [T1] Loaded multi-csv file: 45484 sub-files
INFO [T1] Loading textures...
INFO [T1] Loading sounds...
INFO [T1] Loading time  [data]: 4.368 s
WARN Cannot enter editor mode without a game
WARN Cannot enter editor mode without a game
INFO gaia has civilisation Gaia
INFO name1 has civilisation British
INFO name2 has civilisation French
WARN cannot play: Traceback (most recent call last):
  File ?, in QV4::Runtime::method_callProperty(QV4::ExecutionEngine*, int, QV4::CallData*)+0x2ed [0x7faf3ec1764d]
  File ?, in QV4::QObjectMethod::callInternal(QV4::CallData*, QV4::Scope&) const+0x3c9 [0x7faf3ebffdb9]
  File ?, in ? [0x7faf3ebff0a2]
  File ?, in ? [0x7faf3ebfdb9b]
  File ?, in ? [0x7faf3ec7f5a9]
  File ?, in openage::gui::GameCreator::qt_metacall(QMetaObject::Call, int, void**)+0x45 [0x7faf4151b375]
  File ?, in ? [0x7faf41518862]
  File ?, in openage::gui::GameCreator::activate()+0x23b [0x7faf413f93f3]
  File ?, in qtsdl::GameLogicCaller::in_game_logic_thread(std::function<void ()> const&) const+0x35 [0x7faf41517791]
  File ?, in QMetaObject::activate(QObject*, int, int, void**)+0x97f [0x7faf3ddf16cf]
  File ?, in QtPrivate::QSlotObject<void (qtsdl::GuiCallback::*)(std::function<void ()> const&), QtPrivate::List<std::function<void ()> const&>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)+0x4e [0x7faf4140d404]
  File ?, in qtsdl::GuiCallback::process(std::function<void ()> const&)+0xc [0x7faf4140e08c]
  File ?, in std::function<void ()>::operator()() const+0xe [0x7faf4140e0e6]
  File ?, in ? [0x7faf413f8f6a]
  File ?, in openage::Generator::create(std::shared_ptr<openage::GameSpec>)+0x2fd [0x7faf414372b5]
  File ?, in openage::GameMain::GameMain(openage::Generator const&)+0x411 [0x7faf414246d1]
  File ?, in openage::Generator::add_units(openage::GameMain&) const+0x23d [0x7faf41435941]
  File ?, in openage::UnitContainer::new_unit(openage::UnitType&, openage::Player&, openage::coord::phys3)+0xe8 [0x7faf414dd5bc]
  File ?, in openage::BuildingProducer::place(openage::Unit*, std::shared_ptr<openage::Terrain>, openage::coord::phys3) const+0x4ef [0x7faf414c9adf]
  File ?, in openage::Sound::play() const+0xd0 [0x7faf41427d8c]
  File ?, in openage::audio::Sound::play()+0x5c [0x7faf4138824a]
  File ?, in openage::audio::DynamicResource::use()+0x348 [0x7faf4138104a]
  File ?, in openage::audio::DynamicLoader::create(openage::util::Path const&, openage::audio::format_t)+0x45 [0x7faf413802c9]
  File ?, in openage::audio::OpusDynamicLoader::OpusDynamicLoader(openage::util::Path const&)+0x3f [0x7faf41386147]
  File ?, in openage::audio::open_opus_file(openage::util::Path const&)+0x826 [0x7faf413874f3]
  File ?, in openage::audio::Error::Error(openage::log::message const&)+0x13 [0x7faf4138579b]
  File ?, in openage::error::Error::Error(openage::log::message const&, bool, bool)+0x17e [0x7faf41421258]
  File "libopenage/audio/opus_loading.cpp", line 118, in openage::audio::opus_file_t openage::audio::open_opus_file(const openage::util::Path&)
openage::audio::Error: Could not open opus file: Path(Union(Directory(/home/felix/Desktop/openage/cfg).root @ (b'cfg',), Directory(/home/felix/Desktop/openage/assets).root @ (b'assets',)):/assets/converted/sounds/5123.opus) = '/home/felix/Desktop/openage/assets/converted/sounds/5123.opus': Data didn't contain opus stream
WARN cannot play: Traceback (most recent call last):
  File ?, in QV4::Runtime::method_callProperty(QV4::ExecutionEngine*, int, QV4::CallData*)+0x2ed [0x7faf3ec1764d]
  File ?, in QV4::QObjectMethod::callInternal(QV4::CallData*, QV4::Scope&) const+0x3c9 [0x7faf3ebffdb9]
  File ?, in ? [0x7faf3ebff0a2]
  File ?, in ? [0x7faf3ebfdb9b]
  File ?, in ? [0x7faf3ec7f5a9]
  File ?, in openage::gui::GameCreator::qt_metacall(QMetaObject::Call, int, void**)+0x45 [0x7faf4151b375]
  File ?, in ? [0x7faf41518862]
  File ?, in openage::gui::GameCreator::activate()+0x23b [0x7faf413f93f3]
  File ?, in qtsdl::GameLogicCaller::in_game_logic_thread(std::function<void ()> const&) const+0x35 [0x7faf41517791]
  File ?, in QMetaObject::activate(QObject*, int, int, void**)+0x97f [0x7faf3ddf16cf]
  File ?, in QtPrivate::QSlotObject<void (qtsdl::GuiCallback::*)(std::function<void ()> const&), QtPrivate::List<std::function<void ()> const&>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)+0x4e [0x7faf4140d404]
  File ?, in qtsdl::GuiCallback::process(std::function<void ()> const&)+0xc [0x7faf4140e08c]
  File ?, in std::function<void ()>::operator()() const+0xe [0x7faf4140e0e6]
  File ?, in ? [0x7faf413f8f6a]
  File ?, in openage::Generator::create(std::shared_ptr<openage::GameSpec>)+0x2fd [0x7faf414372b5]
  File ?, in openage::GameMain::GameMain(openage::Generator const&)+0x411 [0x7faf414246d1]
  File ?, in openage::Generator::add_units(openage::GameMain&) const+0x294 [0x7faf41435998]
  File ?, in openage::UnitContainer::new_unit(openage::UnitType&, openage::Player&, openage::coord::phys3)+0xe8 [0x7faf414dd5bc]
  File ?, in openage::LivingProducer::place(openage::Unit*, std::shared_ptr<openage::Terrain>, openage::coord::phys3) const+0x63 [0x7faf414cba99]
  File ?, in openage::MovableProducer::place(openage::Unit*, std::shared_ptr<openage::Terrain>, openage::coord::phys3) const+0x63 [0x7faf414cb9e3]
  File ?, in openage::ObjectProducer::place(openage::Unit*, std::shared_ptr<openage::Terrain>, openage::coord::phys3) const+0x573 [0x7faf414cb8a7]
  File ?, in openage::Sound::play() const+0xd0 [0x7faf41427d8c]
  File ?, in openage::audio::Sound::play()+0x5c [0x7faf4138824a]
  File ?, in openage::audio::DynamicResource::use()+0x348 [0x7faf4138104a]
  File ?, in openage::audio::DynamicLoader::create(openage::util::Path const&, openage::audio::format_t)+0x45 [0x7faf413802c9]
  File ?, in openage::audio::OpusDynamicLoader::OpusDynamicLoader(openage::util::Path const&)+0x3f [0x7faf41386147]
  File ?, in openage::audio::open_opus_file(openage::util::Path const&)+0x826 [0x7faf413874f3]
  File ?, in openage::audio::Error::Error(openage::log::message const&)+0x13 [0x7faf4138579b]
  File ?, in openage::error::Error::Error(openage::log::message const&, bool, bool)+0x17e [0x7faf41421258]
  File "libopenage/audio/opus_loading.cpp", line 118, in openage::audio::opus_file_t openage::audio::open_opus_file(const openage::util::Path&)
openage::audio::Error: Could not open opus file: Path(Union(Directory(/home/felix/Desktop/openage/cfg).root @ (b'cfg',), Directory(/home/felix/Desktop/openage/assets).root @ (b'assets',)):/assets/converted/sounds/5353.opus) = '/home/felix/Desktop/openage/assets/converted/sounds/5353.opus': Data didn't contain opus stream
WARN cannot play: Traceback (most recent call last):
  File ?, in QV4::Runtime::method_callProperty(QV4::ExecutionEngine*, int, QV4::CallData*)+0x2ed [0x7faf3ec1764d]
  File ?, in QV4::QObjectMethod::callInternal(QV4::CallData*, QV4::Scope&) const+0x3c9 [0x7faf3ebffdb9]
  File ?, in ? [0x7faf3ebff0a2]
  File ?, in ? [0x7faf3ebfdb9b]
  File ?, in ? [0x7faf3ec7f5a9]
  File ?, in openage::gui::GameCreator::qt_metacall(QMetaObject::Call, int, void**)+0x45 [0x7faf4151b375]
  File ?, in ? [0x7faf41518862]
  File ?, in openage::gui::GameCreator::activate()+0x23b [0x7faf413f93f3]
  File ?, in qtsdl::GameLogicCaller::in_game_logic_thread(std::function<void ()> const&) const+0x35 [0x7faf41517791]
  File ?, in QMetaObject::activate(QObject*, int, int, void**)+0x97f [0x7faf3ddf16cf]
  File ?, in QtPrivate::QSlotObject<void (qtsdl::GuiCallback::*)(std::function<void ()> const&), QtPrivate::List<std::function<void ()> const&>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)+0x4e [0x7faf4140d404]
  File ?, in qtsdl::GuiCallback::process(std::function<void ()> const&)+0xc [0x7faf4140e08c]
  File ?, in std::function<void ()>::operator()() const+0xe [0x7faf4140e0e6]
  File ?, in ? [0x7faf413f8f6a]
  File ?, in openage::Generator::create(std::shared_ptr<openage::GameSpec>)+0x2fd [0x7faf414372b5]
  File ?, in openage::GameMain::GameMain(openage::Generator const&)+0x411 [0x7faf414246d1]
  File ?, in openage::Generator::add_units(openage::GameMain&) const+0x90 [0x7faf41435794]
  File ?, in openage::UnitContainer::new_unit(openage::UnitType&, openage::Player&, openage::coord::phys3)+0xe8 [0x7faf414dd5bc]
  File ?, in openage::LivingProducer::place(openage::Unit*, std::shared_ptr<openage::Terrain>, openage::coord::phys3) const+0x63 [0x7faf414cba99]
  File ?, in openage::MovableProducer::place(openage::Unit*, std::shared_ptr<openage::Terrain>, openage::coord::phys3) const+0x63 [0x7faf414cb9e3]
  File ?, in openage::ObjectProducer::place(openage::Unit*, std::shared_ptr<openage::Terrain>, openage::coord::phys3) const+0x573 [0x7faf414cb8a7]
  File ?, in openage::Sound::play() const+0xd0 [0x7faf41427d8c]
  File ?, in openage::audio::Sound::play()+0x5c [0x7faf4138824a]
  File ?, in openage::audio::DynamicResource::use()+0x348 [0x7faf4138104a]
  File ?, in openage::audio::DynamicLoader::create(openage::util::Path const&, openage::audio::format_t)+0x45 [0x7faf413802c9]
  File ?, in openage::audio::OpusDynamicLoader::OpusDynamicLoader(openage::util::Path const&)+0x3f [0x7faf41386147]
  File ?, in openage::audio::open_opus_file(openage::util::Path const&)+0x826 [0x7faf413874f3]
  File ?, in openage::audio::Error::Error(openage::log::message const&)+0x13 [0x7faf4138579b]
  File ?, in openage::error::Error::Error(openage::log::message const&, bool, bool)+0x17e [0x7faf41421258]
  File "libopenage/audio/opus_loading.cpp", line 118, in openage::audio::opus_file_t openage::audio::open_opus_file(const openage::util::Path&)
openage::audio::Error: Could not open opus file: Path(Union(Directory(/home/felix/Desktop/openage/cfg).root @ (b'cfg',), Directory(/home/felix/Desktop/openage/assets).root @ (b'assets',)):/assets/converted/sounds/6042.opus) = '/home/felix/Desktop/openage/assets/converted/sounds/6042.opus': Data didn't contain opus stream

FATAL: terminate has been called

uncaught exception

Traceback (most recent call last):
  File ?, in ? [0x7faf472ce6db]
  File ?, in ? [0x7faf3f9abd69]
  File ?, in ? [0x7faf3f937cbc]
  File ?, in ? [0x7faf3f8f0c75]
  File ?, in ? [0x7faf4137d41d]
  File ?, in openage::audio::AudioManager::audio_callback(short*, int)+0x67 [0x7faf4137d381]
  File ?, in openage::audio::SoundImpl::mix_audio(int*, int)+0x50 [0x7faf4138809c]
  File ?, in openage::audio::DynamicResource::get_data(unsigned long, unsigned long)+0xde0 [0x7faf41383ae0]
  File ?, in openage::audio::DynamicResource::load_chunk_async(std::shared_ptr<openage::audio::chunk_info_t>, unsigned long)+0x136 [0x7faf413816b2]
  File ?, in openage::error::Error::Error(openage::log::message const&, bool, bool)+0x17e [0x7faf41421258]
  File "libopenage/audio/dynamic_resource.cpp", line 180, in void openage::audio::DynamicResource::load_chunk_async(std::shared_ptr<openage::audio::chunk_info_t>, size_t)
openage::error::Error: tried to load chunk without audio loader!

current stack:

Traceback (most recent call last):
  File ?, in ? [0x7faf472ce6db]
  File ?, in ? [0x7faf3f9abd69]
  File ?, in ? [0x7faf3f937cbc]
  File ?, in ? [0x7faf3f8f0c75]
  File ?, in ? [0x7faf4137d41d]
  File ?, in openage::audio::AudioManager::audio_callback(short*, int)+0x67 [0x7faf4137d381]
  File ?, in openage::audio::SoundImpl::mix_audio(int*, int)+0x50 [0x7faf4138809c]
  File ?, in openage::audio::DynamicResource::get_data(unsigned long, unsigned long)+0xde0 [0x7faf41383ae0]
  File ?, in openage::audio::DynamicResource::load_chunk_async(std::shared_ptr<openage::audio::chunk_info_t>, unsigned long)+0x1f6 [0x7faf41381772]
  File ?, in ? [0x7faf47db8c74]
  File ?, in ? [0x7faf47db8a41]
  File ?, in ? [0x7faf47db8a06]
  File ?, in openage::error::terminate_handler()+0x19d [0x7faf41421b52]

handing over to the system...

terminate called after throwing an instance of 'openage::error::Error'
  what():  tried to load chunk without audio loader!
Abgebrochen
```

**Comment by TheJJ** [comment]
The real error seems to be:
```
'/home/felix/Desktop/openage/assets/converted/sounds/5123.opus': Data didn't contain opus stream
```

Please use `file` to check for the file type.
Can you use `mpv` or some other media player to play back this sound?


Nevertheless, we have to be more rubust against such failures.

**Comment by k3x** [comment]
```
$ file /home/felix/Desktop/openage/assets/converted/sounds/5123.opus
/home/felix/Desktop/openage/assets/converted/sounds/5123.opus: Ogg data
```

```
$ mpv /home/felix/Desktop/openage/assets/converted/sounds/5123.opus
Playing: /home/felix/Desktop/openage/assets/converted/sounds/5123.opus
[ffmpeg/demuxer] ogg: Codec not found
[lavf] avformat_open_input() failed
[ffmpeg/demuxer] ogg: Codec not found
[lavf] avformat_open_input() failed
Failed to recognize file format.

Exiting... (Errors when loading file)
```

I tested file on a opus file not related with the openage project and here "Opus audio" is shown which is missing on the sound files created by the asset converter.
```
$ file opussong.opus 
opussong.opus: Ogg data, Opus audio,
```

All files in this directory have the same filetype and the same error message when opened with mpv. VLC is also unable to open them.

**Comment by TheJJ** [comment]
Okay, thanks! Is there a sound in the directory that actually is valid opus? (you can likely do `file *.opus` and observe differences).

**Comment by k3x** [comment]
all files in this dir are "Ogg data"
```
~/Desktop/openage/assets/converted/sounds$ file *
5003.opus:  Ogg data
5011.opus:  Ogg data
[...]:              Ogg data
```

**Comment by k3x** [comment]
this is the smallest one:
https://k3x.eu/files/index.php?dl=2018.11.07_11.36.00_5374.opus

**Comment by k3x** [comment]
i re-converted the assets an took a closer look to the log:

```
[...]
NFO [py] [3476/3612] interface/53211.slp
INFO [py] [3477/3612] interface/50300.wav
AttributeError: 'float' object has no attribute 'to_bytes'
Exception ignored in: 'openage.convert.opus.opusenc.write_opus_header'
AttributeError: 'float' object has no attribute 'to_bytes'
INFO [py] [3478/3612] interface/50301.wav
AttributeError: 'float' object has no attribute 'to_bytes'
Exception ignored in: 'openage.convert.opus.opusenc.write_opus_header'
AttributeError: 'float' object has no attribute 'to_bytes'
INFO [py] [3479/3612] interface/50302.wav
AttributeError: 'float' object has no attribute 'to_bytes'
Exception ignored in: 'openage.convert.opus.opusenc.write_opus_header'
AttributeError: 'float' object has no attribute 'to_bytes'
INFO [py] [3480/3612] interface/50303.wav
AttributeError: 'float' object has no attribute 'to_bytes'
Exception ignored in: 'openage.convert.opus.opusenc.write_opus_header'
AttributeError: 'float' object has no attribute 'to_bytes'
INFO [py] [3481/3612] interface/50304.wav
AttributeError: 'float' object has no attribute 'to_bytes'
Exception ignored in: 'openage.convert.opus.opusenc.write_opus_header'
AttributeError: 'float' object has no attribute 'to_bytes'
INFO [py] [3482/3612] interface/50305.wav
INFO [py] [3483/3612] interface/50306.wav
[...]
```

So i think the convert of opus files failed.

**Comment by TheJJ** [comment]
Ah, that's a good hint and very likely the reason for the failure. I have a suspicion that I accidentially changed the variable type to float and now it can't be serialized any more. And those exceptions should not be ignored.

**Comment by mic-e** [comment]
I can confirm this bug. The error originates from this line in `opusenc.pyx`:

`buf[10:12] = inopt['pre_skip'].to_bytes(2, 'little')`

**Comment by TheJJ** [comment]
Fixed in #1070
